tests/func/test_data_cloud.py

```diff
@@ -570,3 +572,20 @@ def test_target_remote(tmp_dir, dvc, make_remote):
         "6b18131dc289fd37006705affe961ef8.dir",
         "b8a9f715dbb64fd5c56e7783c6820a61",
     }
+
+
+def test_remote_missing_data_dir(tmp_dir, scm, dvc, make_remote):
+    tmp_dir.dvc_gen({"dir": {"subfile": "file2 content"}}, commit="add dir")
+    with dvc.config.edit() as conf:
+        conf["remote"]["s3"] = {"url": "s3://bucket/name"}
+        conf["core"] = {"remote": "s3"}
+
+    remove("dir")
+    remove(dvc.odb.local.cache_dir)
+
+    with patch(
+        "dvc.data_cloud.DataCloud.get_remote_odb",
+        side_effect=RemoteMissingDepsError("remote missing"),
+    ):
```

Comment:
To maintain consistency between tests I would suggest using `mocker` fixture.
then just `mocker.patch`.

Reply:
8a37f70e